src/main/java/codeu/controller/FeedServlet.java

```
@@ -0,0 +1,153 @@
+
+    /**
+     * Set up state for handling conversation-related requests. This method is only called when
+     * running in a server, not when running in a test.
+     */
+    @Override
+    public void init() throws ServletException {
+        super.init();
+        setUserStore(UserStore.getInstance());
+        setConversationStore(ConversationStore.getInstance());
+        setMessageStore(MessageStore.getInstance());
+    }
+
+    /**
+     * Sets the UserStore used by this servlet. This function provides a common setup method for use
+     * by the test framework or the servlet's init() function.
+     */
+    void setUserStore(UserStore userStore) {
+        this.userStore = userStore;
+    }
+
+    /**
+     * Sets the ConversationStore used by this servlet. This function provides a common setup method
+     * for use by the test framework or the servlet's init() function.
+     */
+    void setConversationStore(ConversationStore conversationStore) {
+        this.conversationStore = conversationStore;
+    }
+
+    /**
+     * This function fires when a user navigates to the conversations page. It gets all of the
+     * conversations from the model and forwards to conversations.jsp for rendering the list.
+     */
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response)
+            throws IOException, ServletException {
+        List<Conversation> conversations = conversationStore.getAllConversations();
+        List<User> users = userStore.getAllUsers();
+        List<Message> messages = messageStore.getAllMessages();
+
+        List<FeedEntry> entries = new ArrayList<FeedEntry>();
+
+        entries.addAll(conversations);
+        entries.addAll(users);
+        entries.addAll(messages);
+
+        Collections.sort(entries, new FeedEntryComporator());
+
+        request.setAttribute("entries", entries);
+        request.getRequestDispatcher("/WEB-INF/view/feed.jsp").forward(request, response);
+    }
+
+    /**
```

Any reason to comment out this code ? I'm guessing you haven't implemented Post request yet, so you can remove the comment, and just throw UnsupportedOperationException

---

Oh yeah I copied another Servlet and I commented out the code I didn't need as I was developing. Looks like I forgot to delete that!